### PR TITLE
[34] Create shapes component

### LINKED ISSUE #34: Create shapes component

### PR DESCRIPTION
**Pull request by n0macx**
Closes #34

Information:

- [x] Do your changes affect most of the application?
  - [x] Did you describe it in the task?
  - [x] Did you inform others about it on Slack?
- [x] Have you installed a new library?
  - [x] Have you consulted it with the others?

Checklist:

- [x] Does the Pull Request title meet the rules? (eg. `[9] Add name's rules for PR to readme`)
- [x] Does the name of your branch meet the rules? (eg. `9-add-names-rules-for-pr-to-readme`)
- [x] Does the Pull Request description contain `Closes #<issue number>` (eg. `Closes #9`)?
- [x] Did Husky run all jobs? (you didn't use the `no-verify` flag)
- [x] Did you describe the task in the linked issue? (AC, useful links, description)
- [x] Did you link the issue to the pull request through the `Development` tab in the pull request options?

Remainders:

- Make sure your commit messages contain useful information
- Make sure to use yarn and do not push other lock files

### CODE REVIEW

File: src/common/globals.css [hunk 1]
```diff
@@ -7,3 +7,14 @@
...
+  -webkit-clip-path: polygon(
+    25% 5%,
+    75% 5%,
+    100% 50%,
+    75% 95%,
+    25% 95%,
+    0% 50%
+  );
+  clip-path: polygon(25% 5%, 75% 5%, 100% 50%, 75% 95%, 25% 95%, 0% 50%);
+}
```
Comment (jedrzejratajczak) [inline src/common/globals.css:20]: I'm not sure about putting it here. Can you check if we can have it closer to the hexagon directory? I don't think we will reuse it anywhere else

File: src/components/Hexagon/lookup.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+export const hexColors = {
```
Comment (jedrzejratajczak) [inline src/components/Hexagon/lookup.tsx:1]: There is no React here. You can rename the file to `lookup.ts`

File: src/components/index.ts [hunk 3]
```diff
@@ -1 +0,0 @@
-export {};
```
Comment (jedrzejratajczak) [inline src/components/index.ts:1]: We need this file for simple imports from `@components`.
Try:
```js
export * from './Hexagon';
```

File: src/components/Hexagon/index.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export const Hexagon = ({ color, opacity, icon, iconOrShadowColor, variant, ...props }: HexProps) => {
+  const colorClasses = hexColors[color];
+  const opacityClasses = hexOpacity[opacity];
+  const iconColorClasses = hexColors[iconOrShadowColor];
+  const shadowClasses = hexShadowColor[iconOrShadowColor];
+
+  return (
+    <div className={`h-full w-full ${variant === 'withShadow' ? shadowClasses : ''}`}>
```
Comment (jedrzejratajczak) [inline src/components/Hexagon/index.tsx:20]: hexVariants[variant]

File: src/components/Hexagon/lookup.tsx (hunk 2 above)
Comment (n0macx) [reply src/components/Hexagon/lookup.tsx:1]: Done.

File: src/components/Icon/index.tsx [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const Icon = (color: any) => {
+  return (
+    <div className={`h-6 w-6 ${color}`} data-testid="icon">
+      Ico
+    </div>
+  );
+};
```
Comment (jedrzejratajczak) [inline src/components/Icon/index.tsx:8]: We could use some TODO comment to replace this component with a proper one. Can also add a link to the task, like:
```ts
// TODO: Replace the component with a proper one in https://github.com/CodersCrew/coderscrew-website/issues/33
```

File: src/components/Hexagon/hexagon.test.tsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    const hexagon = screen.getByTestId('hexagon');
+    expect(hexagon).toBeInTheDocument();
+  });
+
+  test('if hexagon with icon is rendered', () => {
+    render(<Hexagon color="blue" opacity={100} iconOrShadowColor="blue" variant="default" icon />);
+    const icon = screen.getByTestId('icon');
+    expect(icon).toBeInTheDocument();
+  });
+});
```
Comment (jedrzejratajczak) [inline src/components/Hexagon/hexagon.test.tsx:18]: How about this?

```ts
describe('Hexagon', () => {
  it('should render without errors', () => { ... });
  it('should render with icon', () => { ... });
});
```

**Review by jedrzejratajczak — APPROVED** [review]
I made some changes. Resolve my last 2 comments and you can merge it

**Pull request merged**